bundles/org.openhab.core.model.script/src/org/openhab/core/model/script/lib/NumberExtensions.java

```diff
@@ -380,6 +380,24 @@ public static BigDecimal numberToBigDecimal(Number number) {
             }
             return null;
         }
+        if (number instanceof DecimalType dtype) {
+            return dtype.toBigDecimal();
+        }
+        if (number instanceof BigDecimal decimal) {
+            return decimal;
+        }
+        if (number instanceof Double value) {
+            return new BigDecimal(value);
+        }
+        if (number instanceof Float value) {
+            return new BigDecimal(value);
+        }
+        if (number instanceof Long value) {
+            return new BigDecimal(value);
```

Comment:
For type long maybe better use valueOf(long), which should allow for internal caching mechanisms.
See API note here https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/math/BigDecimal.html#valueOf(long)

Reply:
Thanks for the hint, will change this.